src/relentless/simulate/initialize.py

```diff
@@ -123,11 +124,16 @@ def _pack_particles(seed, N, V, diameters):
         types = []
         trees = {}
         Nadded = 0
+        evaluated_diameters = copy.deepcopy(diameters)
+        for keys in evaluated_diameters:
+            evaluated_diameters[keys] = variable.evaluate(evaluated_diameters[keys])
         # insert the particles, big to small
-        sorted_N = sorted(N.items(), key=lambda x: x[1], reverse=True)
-        for i, Ni in sorted_N:
+        sorted_diameters = sorted(
+            evaluated_diameters.items(), key=lambda x: x[1], reverse=True
+        )
```

Comment:
I think you can use a generator comprehension in `sorted`, as `N.items()` should be a generator of tuples:
```suggestion
        sorted_diameters = sorted(
            (i, variable.evaluate(diameters[i])) for i in N,
            key=lambda x: x[1],
            reverse=True
        )
```
Then you would not need the stuff above.